eos/eos.hpp

```diff
@@ -448,6 +448,137 @@ template <typename T> class RelativisticEOS {
   Real cl_, cl2_;
 };
 
+template <typename T> class EAPRampEOS {
+  
+  // calculate alpha given a particular pressure
+  PORTABLE_FUNCTION
+  Real alpha_p(const Real P) const {
+    if (P < Pe_) {
+      return alpha0_;
+    } else if (P > Pc_) {
+      return 1.0;
+    } else {
+      return alpha0_ + (P - Pe_) / (Pc_ - Pe_) * (1.0 - alpha0_);
+    }
+  }
+  
+  template<typename F, typename ... ARGS>
+  PORTABLE_FUNCTION
+  auto alpha_impl(F func, const Real rho, ARGS... args) const {
```

Comment:
I think you dropped this `&&`

Reply:
I added it, and I'm using lambdas now instead of function pointers, I think this should be easier for the compiler to inline and optimize, plus function pointers are gross.